plugins/view-resources/src/utils.ts

```diff
@@ -401,13 +402,39 @@ export async function deleteObject (client: TxOperations, object: Doc): Promise<
   }
 }
 
+export async function getCurrentPersonAccounts (): Promise<Set<Ref<Account> | undefined>> {
+  return new Set(
+    (
+      await getClient().findAll(contact.class.PersonAccount, { person: (getCurrentAccount() as PersonAccount).person })
+    ).map((it) => it._id)
+  )
+}
+
 export async function deleteObjects (client: TxOperations, objects: Doc[], skipCheck: boolean = false): Promise<void> {
+  let realObjects: Doc[] = []
   if (!skipCheck) {
     const currentAcc = getCurrentAccount()
-    if (currentAcc.role !== AccountRole.Owner && objects.some((p) => p.createdBy !== currentAcc._id)) return
+
+    // We need to find all person current accounts
+    const allPersonAccounts = await getCurrentPersonAccounts()
+
+    const byClass = new Map<Ref<Class<Doc>>, Doc[]>()
+    for (const d of objects) {
+      byClass.set(d._class, [...(byClass.get(d._class) ?? []), d])
+    }
+    for (const [cl, docs] of byClass.entries()) {
+      const realDocs = await client.findAll(cl, { _id: { $in: docs.map((it: Doc) => it._id) } })
+      const notAllowed = realDocs.filter((p) => !allPersonAccounts.has(p.createdBy))
+      if (currentAcc.role !== AccountRole.Owner && notAllowed.length > 0) {
+        console.error('You are not allowed to delete this object', notAllowed)
+      }
+      realObjects.push(...realDocs.filter((p) => allPersonAccounts.has(p.createdBy)))
```

Comment:
if the current account is Owner, then `realObjects` might contain not all objects, we should probably omit this check for owners at all.

Reply:
Thanks, will fix